Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -62,3 +65,41 @@ public class OpenFeatureAPI {
         self.hooks.removeAll()
     }
 }
+
+// MARK: Provider Events
+
+extension OpenFeatureAPI {
+
+    public func addHandler(observer: Any, selector: Selector, event: ProviderEvent) {
+        providerNotificationCentre.addObserver(
+            observer,
+            selector: selector,
+            name: event.notification,
+            object: nil
+        )
+    }
+
+    public func removeHandler(observer: Any, event: ProviderEvent) {
+        providerNotificationCentre.removeObserver(observer, name: event.notification, object: nil)
+    }
+
+    public func emitEvent(
+        _ event: ProviderEvent,
+        provider: FeatureProvider,
+        error: Error? = nil,
+        details: [AnyHashable: Any]? = nil
+    ) {
+        var userInfo: [AnyHashable: Any] = [:]
+        userInfo[ProviderEventDetailsKeyProvider] = provider
+
+        if let error {
+            userInfo[ProviderEventDetailsKeyError] = error
+        }
+
+        if let details {
+            userInfo.merge(details) { $1 } // Merge, keeping value from `details` if any conflicts
+        }
+
+        providerNotificationCentre.post(name: event.notification, object: nil, userInfo: userInfo)
```

Comment:
why do we need the userinfo?

Reply:
User info is just representation of additional information about the event in form of a dictionary. i.e. information about client, provider, error etc. According to specification I think it’s not required (didn’t find any references to it), but from developer experience - I believe it’s nice to have.

Reply:
There is a requirement to provide the client name for client / provider events, but not for adding the provider details for API events. I figured there's no harm in providing both, it's also generally expected in iOS to fill the `userInfo` with these kind of things.